### PR TITLE
introduce wildfly image documentation

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
@openshift/team-documentation ptal, can merge after https://github.com/openshift-s2i/s2i-wildfly/pull/122

**Comment by bparees** [comment]
note: origin only.

File: _topic_map.yml [hunk 1]
```diff
@@ -834,6 +834,8 @@ Topics:
         File: python
       - Name: Ruby
         File: ruby
+      - Name: Wildfly
+        File: wildfly
```
Comment (adellape) [inline _topic_map.yml]: Add if this is for just Origin:

```
Distros: openshift-origin
```

File: using_images/s2i_images/wildfly.adoc [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+These images are available on DockerHub. To download them:
+
+----
+$ docker pull openshift/wildfly-101-centos7
+$ docker pull openshift/wildfly-100-centos7
+----
+
+To use these images, you can either access them directly from these
+xref:../../architecture/infrastructure_components/image_registry.adoc#architecture-infrastructure-components-image-registry[image
+registries], or push them into your
```
Comment (adellape) [inline using_images/s2i_images/wildfly.adoc]: I know this part is just taken from the other image topics, but the "these image registries" part seems weird/vague. I guess it's referring to the pull specs that precede it? If so then I guess it's just the link that makes it confusing.

File: using_images/s2i_images/wildfly.adoc [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+|===
+
+|Variable name |Description
+
+|`*MAVEN_ARGS*`
+|Overrides the default arguments passed to maven during the build process.
+
+|`*MAVEN_ARGS_APPEND*`
+|This value will be appended to either the default maven arguments, or the value of MAVEN_ARGS if 
+MAVEN_ARGS is set.
```
Comment (adellape) [inline using_images/s2i_images/wildfly.adoc]: Markup `MAVEN_ARGS` x2

File: using_images/s2i_images/wildfly.adoc [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+|`*MAVEN_OPTS*`
+|Contains JVM parameters to maven.  Will be appended to JVM arguments that are calculated by the image
```
Comment (adellape) [inline using_images/s2i_images/wildfly.adoc]: s/maven/Maven/ ?

Comment (adellape) [inline using_images/s2i_images/wildfly.adoc]: Throughout.

File: using_images/s2i_images/wildfly.adoc [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+|`*CONTAINER_CORE_LIMIT*`
+|When set to a non-null value, the number of parallel garbage collection threads will be set to this value.
+
+|`*AUTO_DEPLOY_EXPLODED*`
+|When set to `true`, Wildfly will automatically deploy exploded war content.  When unset or set to `false`, 
+a `.dodeploy` file must be touched to trigger deployment of exploded war content.
+
+|`*MYSQL_DATABASE*`
+|If set, WildFly will attempt to define a MySQL datasource based on the assumption you have an OpenShift service named "mysql" defined.
```
Comment (adellape) [inline using_images/s2i_images/wildfly.adoc]: Markup:

```
`mysql`
```

File: using_images/s2i_images/wildfly.adoc [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+It will attempt to reference the following environment variables which are automatically defined if the `mysql` service exists:
+`MYSQL_SERVICE_PORT`, `MYSQL_SERVICE_HOST`, `MYSQL_PASSWORD`, and `MYSQL_USER`.
+
+|`*POSTGRESQL_DATABASE*`
+|If set, WildFly will attempt to define a PostgreSQL datasource based on the assumption you have an OpenShift service named "postgresql" defined.
```
Comment (adellape) [inline using_images/s2i_images/wildfly.adoc]: Markup:

```
`postgresql`
```

**Review by adellape — COMMENTED** [review]
Comments inline, otherwise LGTM.

File: using_images/s2i_images/wildfly.adoc [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+----
+
+To use these images, you can either access them directly from these
+xref:../../architecture/infrastructure_components/image_registry.adoc#architecture-infrastructure-components-image-registry[image
+registries], or push them into your
+xref:../../architecture/infrastructure_components/image_registry.adoc#integrated-openshift-registry[{product-title}
+Docker registry]. Additionally, you can create an
+xref:../../architecture/core_concepts/builds_and_image_streams.adoc#image-streams[image
+stream] that points to the image, either in your Docker registry or at the
+external location. Your {product-title}t resources can then reference the
```
Comment (adellape) [inline using_images/s2i_images/wildfly.adoc]: Stray `t`

File: using_images/s2i_images/wildfly.adoc [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+ifdef::openshift-enterprise[]
+The Wildfly S2I builder image
+endif::openshift-enterprise[]
+assembles your application source with any required dependencies to create a
+new image containing your Java application. This resulting image can be run
+either by {product-title} or by Docker.
+
+== Versions
+Currently, {product-title} supports version
+https://github.com/openshift-s2i/s2i-wildfly/tree/master/10.1[10.1], and
```
Comment (adellape) [inline using_images/s2i_images/wildfly.adoc]: Stray comma.

Comment (bparees) [inline using_images/s2i_images/wildfly.adoc]: anticipatory oxford comma :)

File: using_images/s2i_images/wildfly.adoc (hunk 2 above)
Comment (bparees) [inline using_images/s2i_images/wildfly.adoc]: will reword, at least in this case it's simpler since they only exist on dockerhub.

**Comment by bparees** [comment]
@adellape updated

**Comment by bparees** [comment]
@adellape pre-req PRs have merged, this should be good to go (I think i addressed all your comments)

**Comment by adellape** [comment]
:+1: Thanks.

**Comment by adellape** [comment]
[Document: Tune maven memory limits in wildfly image [techdebt]](https://trello.com/c/gtUxCrC5/508-document-tune-maven-memory-limits-in-wildfly-image-techdebt)

**Comment by adellape** [comment]
[Document: Support exploded war hotdeploy for Wildfly [techdebt]](https://trello.com/c/106uJZ1P/507-document-support-exploded-war-hotdeploy-for-wildfly-techdebt)

**Pull request merged**